junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/JupiterTestDescriptor.java

```diff
@@ -88,9 +97,33 @@
 		// @formatter:on
 	}
 
-	protected static <E extends AnnotatedElement> String determineDisplayName(E element,
-			Function<E, String> defaultDisplayNameGenerator) {
+	protected static DisplayNameGenerator getDisplayNameGenerator(Class<?> testClass) {
+		Preconditions.notNull(testClass, "Test class must not be null");
+		// first locate direct or meta-direct present annotations
+		DisplayNameGeneration generation = findAnnotation(testClass, DisplayNameGeneration.class).orElse(null);
+		// next, crawl enclosing classes
+		if (generation == null) {
+			Class<?> current = testClass.getEnclosingClass();
+			while (current != null) {
+				DisplayNameGeneration enclosedGeneration = current.getAnnotation(DisplayNameGeneration.class);
```

Comment:
We cannot use `getAnnotation()` since it will not pick up meta-annotations.

Please use `AnnotationUtils.findAnnotation()` instead.